sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -830,6 +849,142 @@ def init_port_sfp_status_tbl(port_mapping, stop_event=threading.Event()):
+            while not self.task_stopping_event.is_set():
+                # Handle port change event from main thread
+                port_mapping.handle_port_config_change(sel, asic_context, self.task_stopping_event, self.port_mapping, helper_logger, self.on_port_config_change)
+        else:
+            while not self.task_stopping_event.is_set():
+                try:
+                    msg = self.task_queue.get(False, 0.1)
+                except:
+                    msg = None
+                if msg is None:
+                    time.sleep(1)
+                    continue
+
+                lport = msg['lport']
+                pport = msg['pport']
+                lanes = msg['lanes']
+                speed = msg['speed']
+                if speed == 0 or len(lanes) < 1:
+                    continue
+                sfp = platform_chassis.get_sfp(pport)
+                if not sfp.get_presence():
+                    continue
+
+                try:
+                    # Skip if the port/cage type is not QSFP-DD
+                    if sfp.SFP_PORT_TYPE_QSFPDD != sfp.get_port_type():
+                        continue
+                    # Skip if the xcvr type is not QSFP-DD
+                    info = sfp.get_transceiver_info()
+                    if info.get('type_abbrv_name', None) not in ['QSFP-DD', 'QSFP_DD']:
+                        continue
+                    # Skip if the memory type is flat
+                    if info.get('memory_type', 'flat') in ['flat', 'FLAT']:
+                        continue
+
+                    # Fetch the custom CMIS media setting from media_settings.json
+                    conf = get_media_settings_cmis(info)
+                    if conf is None:
+                        conf = {}
+                    # Enable application selection by default
+                    if conf.get('application_selection', True):
+                        self.log_notice("{}: application update for {}G, {}-lanes".format(
+                            lport, int(speed/1000), len(lanes)))
+                        ret = sfp.set_cmis_application(speed, len(lanes))
+                    else:
+                        ret = None
+                except (NotImplementedError, AttributeError):
+                    ret = None
+
+                if ret is None:
+                    state = "skipped"
+                elif not ret:
+                    state = "failed"
+                else:
+                    state = "ok"
+
+                self.log_notice("{}: application update ... {}".format(lport, state))
+
+        self.log_notice("Stopped")
+
+    def task_run(self, y_cable_presence):
+        if platform_chassis is None:
+            self.log_notice("Platform chassis is not available, stopping...")
+            return
+
+        has_cmis = False
+        for sfp in platform_chassis.get_all_sfps():
+            if sfp.get_port_type() in [sfp.SFP_PORT_TYPE_QSFPDD]:
```

Comment:
more than just QSFPDD supports CMIS

Reply:
As of now, only QSFPDD (type=18h) is verified, we could circle back on OSFP(type=19h) once the sample is available

Reply:
Just for a record, the get_port_type() checker is already removed in the latest commit